cmd/istioctl/main.go

```diff
@@ -318,10 +346,12 @@ Example usage:
 func init() {
 	rootCmd.PersistentFlags().StringVarP(&kubeconfig, "kubeconfig", "c", "",
 		"Use a Kubernetes configuration file instead of in-cluster configuration")
-	rootCmd.PersistentFlags().StringVarP(&managerAddr, "managerAddr", "m", "",
-		"Set your Istio manager address")
 	rootCmd.PersistentFlags().StringVarP(&namespace, "namespace", "n", api.NamespaceDefault,
 		"Select a Kubernetes namespace")
+	rootCmd.PersistentFlags().StringVar(&istioNamespace, "istioNamespace", defaultIstioNamespace,
+		"Namespace where Istio system resides")
+	rootCmd.PersistentFlags().IntVar(&apiServerPort, "port", 8081,
+		"Manager API server port")
 
```

Comment:
I agree that we should decouple from k8s before it gets even harder to accomplish down the road. That means no k8s dependencies in base istioctl, and the only functional dependency being authentication via JWT token to kube API server proxy. If this is too hard, I'd prefer creating our own JWT validation and using a load balancer IP for the manager config server.